File: src/main/java/factory/LibraryServiceFactory.java
package factory;

import services.LibraryService;
import services.MavenCentralLibraryService;
import services.NpmRegistryLibraryService;
import services.PypiLibraryService;

public class LibraryServiceFactory {

    public LibraryService createLibraryService(String provider) {
        switch (provider) {
            case "maven":
            case "gradle":
                return new MavenCentralLibraryService();
            case "npm":
                return new NpmRegistryLibraryService();
            case "pypi":
                return new PypiLibraryService();
            default:
                return null;
        }
    }
}

File: src/main/java/factory/LibraryServiceFactory.kt
package factory

import services.age.LibraryService
import services.age.MavenCentralLibraryService
import services.age.NpmRegistryLibraryService
import services.age.PypiLibraryService

fun createLibraryService(provider: String?): LibraryService? {
    return when (provider) {
        "maven", "gradle" -> MavenCentralLibraryService()
        "npm" -> NpmRegistryLibraryService()
        "pypi" -> PypiLibraryService()
        else -> null
    }
}
